sycl/include/sycl/detail/generic_type_traits.hpp

```diff
@@ -380,15 +380,49 @@ template <typename T> class TryToGetPointerVecT {
   using type = decltype(check(T()));
 };
 
+// IsNonLegacyMultiPtr
+template <typename T> struct IsNonLegacyMultiPtr {
+  static constexpr bool value = false;
+};
+
+template <typename ElementType, access::address_space Space>
+struct IsNonLegacyMultiPtr<
+    multi_ptr<ElementType, Space, access::decorated::yes>> {
+  static constexpr bool value = true;
+};
+
+template <typename ElementType, access::address_space Space>
+struct IsNonLegacyMultiPtr<
+    multi_ptr<ElementType, Space, access::decorated::no>> {
+  static constexpr bool value = true;
+};
+
+// IsLegacyMultiPtr
+template <typename T> struct IsLegacyMultiPtr {
+  static constexpr bool value = false;
+};
+
+template <typename ElementType, access::address_space Space>
+struct IsLegacyMultiPtr<
+    multi_ptr<ElementType, Space, access::decorated::legacy>> {
+  static constexpr bool value = true;
+};
+
```

Comment:
I think this should go to somewhere near https://github.com/intel/llvm/blob/114906b688f912f58a2e90901b92ce18db37b999/sycl/include/sycl/detail/type_traits.hpp#L283-L293 with corresponding change in naming convention. Also, I think it would be better to introduce a simpler `IsMultiPtr` together with these more fine-grained traits.

I'd also like to see `is_*multi_ptr_v` helpers to be implemented here.

Reply:
Applied.